### PR TITLE
adds a gitlab config option to remove source branch after PR is merged

### PR DESCRIPTION
**Pull request by guumaster**
Gitlab have a nice feature to remove source branches automatically after a PR is merged. This code enables `renovate` to setup that option on GitLab PRs.

 I've tested it with all the different formats, in config file as `gitlabRemoveSourceBranch`, as a CLI option `--gitlab-remove-source-branch=true` and as an env var `RENOVATE_GITLAB_REMOVE_SOURCE_BRANCH` and all works fine. 

I've put it under the `gitlab` prefix because the same feature is not present on GitHub.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -56,24 +56,25 @@ $ node renovate --help
...
+    --ignore-deps <list>                     Dependencies to ignore
+    --ignore-future [boolean]                Ignore versions tagged as "future"
+    --ignore-unstable [boolean]              Ignore versions with unstable semver
+    --respect-latest [boolean]               Ignore versions newer than npm "latest" version
+    --recreate-closed [boolean]              Recreate PRs even if same ones were closed previously
+    --rebase-stale-prs [boolean]             Rebase stale PRs
+    --labels <list>                          Labels to add to Pull Request
+    --assignees <list>                       Assignees for Pull Request
+    --reviewers <list>                       Requested reviewers for Pull Requests
+    --gitlab-remove-source-branch [boolean]  Remove source branch on merge
```
Comment (guumaster) [inline docs/configuration.md]: All this change is due to more spacing needed for a longer option.

File: lib/config/definitions.js [hunk 2]
```diff
@@ -131,6 +131,12 @@ const options = [
     description: 'Requested reviewers for Pull Requests',
     type: 'list',
   },
+  {
+    name: 'gitlabRemoveSourceBranch',
+    description: 'Remove source branch on merge',
+    type: 'boolean',
+    default: false,
```
Comment (rarkins) [inline lib/config/definitions.js]: What's your opinion on making it default to true? It doesn't sound to me like anyone should object to this change in behaviour as they can until the option manually as well, right?

Comment (guumaster) [inline lib/config/definitions.js]: Great, I'd also  prefer it as a default behaviour. I'll update it

Comment (guumaster) [inline lib/config/definitions.js]: done.

**Comment by guumaster** [comment]
pushed again with updated README.md too.

**Comment by rarkins** [comment]
@guumaster I'm considering whether to remove the `gitlab-` prefix or not. Already there are a few configuration options which are *GitHub*-only (e.g. reviewers), and I don't wish to rename them too. I'm wondering whether marking each as "GitHub only" or "GitLab" only in the description/docs would be sufficient.

Arguments for using prefix:
- Makes it very clear not to use it with GitHub

Arguments against using prefix:
- Makes configuration options longer!
- Existing GitHub-only options are not prefixed currently
- If we were to support a third provider in future (e.g. Bitbucket) and options worked on 2 of 3 providers, how could we prefix them then? Answer: we probably couldn't

Based on the above, I'm thinking the answer may be to remove this gitlab prefix and then also clarify any existing GitHub-only options in documentation/description. What do you think?

**Comment by rarkins** [comment]
I've updated existing descriptions in #129 but won't merge it yet as it will cause this branch to have merge conflicts

**Comment by guumaster** [comment]
You are right. Its not pretty to have prefixed options and it'd be cumbersome to maintain. It also could be checked internally and show a warning according your platform and options.

I'll remove prefix and push it again. You can also merge #129, I'll rebase and fix this one if needed.

**Comment by rarkins** [comment]
@guumaster I decided to make this default and non-configurable behaviour. I can't think of any good reason you'd want to keep renovate branches around anyway. Thanks for the suggested feature. Now published in 8.9.0

**Pull request closed without merging**